beacon-chain/db/kv/lightclient.go

```diff
@@ -0,0 +1,65 @@
+package kv
+
+import (
+	"context"
+	"encoding/binary"
+	ethpbv2 "github.com/prysmaticlabs/prysm/v5/proto/eth/v2"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+func (s *Store) SaveLightClientUpdate(ctx context.Context, period uint64, update *ethpbv2.LightClientUpdateWithVersion) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.saveLightClientUpdate")
+	defer span.End()
+
+	return s.db.Update(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(lightClientUpdatesBucket)
+		updateMarshalled, err := encode(ctx, update)
+		if err != nil {
+			return err
+		}
+		return bkt.Put(uint64ToBytes(period), updateMarshalled)
+	})
+}
+
+func (s *Store) LightClientUpdates(ctx context.Context, startPeriod, endPeriod uint64) ([]*ethpbv2.LightClientUpdateWithVersion, error) {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.LightClientUpdates")
+	defer span.End()
+
+	var updates []*ethpbv2.LightClientUpdateWithVersion
```

Comment:
This is OK, but as a small optimization you can pre-allocate the capacity of the slice, since you know the slice won't be bigger than the number of periods.
```
updates := make([]*ethpbv2.LightClientUpdateWithVersion, 0, endPeriod - startPeriod + 1)
```